### PR TITLE
Upgrade graphql to 14.6.0

### PR DESCRIPTION
**Pull request by vsumner**
Upgrades `graphql` from `14.2.0` to `14.6.0` and removes `@types/graphql` because `graphql` exports its own types since `14.5.0`.

so many 🐛 fixes: [14.6.0](https://github.com/graphql/graphql-js/releases/tag/v14.6.0)

Some nice new features [14.5.0](https://github.com/graphql/graphql-js/releases/tag/v14.5.0)

- Add 'extensions' to all Type System objects

More new features [14.4.0](https://github.com/graphql/graphql-js/releases/tag/v14.4.0)

- Use 'Object.freeze' consistently on all exported Array/Object constants

This PR also updates `apollo-codegen-core` from `0.35.0` to `0.36.4` so that the same version of graphql can be used. https://github.com/apollographql/apollo-tooling/blob/master/CHANGELOG.md

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4453,12 +4515,17 @@ lodash.unescape@4.0.1:
...
+  resolved "https://registry.yarnpkg.com/lodash.xorby/-/lodash.xorby-4.7.0.tgz#9c19a6f9f063a6eb53dd03c1b6871799801463d7"
+  integrity sha1-nBmm+fBjputT3QPBtocXmYAUY9c=
+
+lodash@^4.11.1, lodash@^4.17.11, lodash@^4.2.1, lodash@^4.3.0:
   version "4.17.11"
   resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.11.tgz#b39ea6229ef607ecd89e2c8df12536891cac9b8d"
   integrity sha512-cQKh8igo5QUhZ7lg38DYWAxMvjSAKG0A8wGSVimP07SIUEK2UO+arSRKbRZWtelMtN5V0Hkwh5ryOto/SshYIg==
 
-lodash@^4.17.12, lodash@^4.17.13, lodash@^4.17.14:
+lodash@^4.17.12, lodash@^4.17.13, lodash@^4.17.14, lodash@^4.17.4:
```
Comment (alexandcote) [inline yarn.lock:4528]: Can you dedupe `lodash` here :)

File: yarn.lock [hunk 2]
```diff
@@ -3011,7 +3069,18 @@ graceful-fs@^4.1.11, graceful-fs@^4.1.15, graceful-fs@^4.1.2, graceful-fs@^4.1.6
...
+  resolved "https://registry.yarnpkg.com/graphql-config/-/graphql-config-2.2.2.tgz#a4b577826bba9b83e7b0f6cd617be43ca67da045"
+  integrity sha512-mtv1ejPyyR2mJUUZNhljggU+B/Xl8tJJWf+h145hB+1Y48acSghFalhNtXfPBcYl2tJzpb+lGxfj3O7OjaiMgw==
+  dependencies:
+    graphql-import "^0.7.1"
+    graphql-request "^1.5.0"
+    js-yaml "^3.10.0"
+    lodash "^4.17.4"
+    minimatch "^3.0.4"
+
+graphql-config@^2.0.1:
```
Comment (alexandcote) [inline yarn.lock:3083]: Dedupe

File: yarn.lock [hunk 3]
```diff
@@ -1758,6 +1802,11 @@ core-js@^3.0.1:
   resolved "https://registry.yarnpkg.com/core-js/-/core-js-3.2.1.tgz#cd41f38534da6cc59f7db050fe67307de9868b09"
   integrity sha512-Qa5XSVefSVPRxy2XfUC13WbvqkxhkwB3ve+pgCQveNgYzbM/UxZeu1dcOX/xr4UmfUd+muuvsaxilQzCyUurMw==
 
+core-js@^3.4.0:
```
Comment (alexandcote) [inline yarn.lock:1805]: Dedupe.

File: yarn.lock [hunk 4]
```diff
@@ -1498,6 +1535,13 @@ combined-stream@^1.0.6, combined-stream@~1.0.6:
   dependencies:
     delayed-stream "~1.0.0"
 
+combined-stream@^1.0.8:
```
Comment (alexandcote) [inline yarn.lock:1538]: Dedupe

**Review by alexandcote — APPROVED** [review]
This PR looks good to me. Small comment otherwise 🚢

**Comment by vsumner** [comment]
Refactored the require `"graphql": ">=14.5.0 <15.0.0"`

File: yarn.lock (hunk 1 above)
Comment (vsumner) [reply yarn.lock]: Screen Shot 2020-04-09 at 12 15 20 PM

**Review by alexandcote — APPROVED** [review]
All this make sense to me 👍

**Pull request merged**